Dynatrace/dynatrace-operator
# Move OneAgent type into its own section

## Description

The goal is to make the code base more modular and maintainable. 
This PR extracts the OneAgent type into its own package.

[DAQ-1368](https://dt-rnd.atlassian.net/browse/DAQ-1368)

## How can this be tested?

- Run unittests and linters => everything should be fine



[DAQ-1368]: https://dt-rnd.atlassian.net/browse/DAQ-1368?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
